aeneas/src/vst/Verifier.v3

```diff
@@ -886,29 +893,61 @@ class VstCompoundVerifier {
 						_ => ;
 					}
 				}
-				if (rep.length > bitWidth) errAtToken(token).BitPatternTooLong(bitWidth, rep.length);
-			}
-			Tuple(list) => {
-				for (l = list.list; l != null; l = l.tail) {
-					checkPackingExpr(env, l.head, MAX_PACKING_WIDTH); // TODO: proper size
-				}
-			}
-			App(p, args) => { // TODO
-				// TODO: resolve and apply packing {p}
-				for (l = args.list; l != null; l = l.tail) {
-					checkPackingExpr(env, l.head, MAX_PACKING_WIDTH); // TODO: proper size
+				return byte.!(rep.length);
+			}
+			Concat(l) => {
+				var total: byte = 0;
+				for (e = l.list; e != null; e = e.tail) {
+					var s = inferPackingExpr(env, e.head);
+					if (s == MAX_PACKING_WIDTH || total == MAX_PACKING_WIDTH) total = MAX_PACKING_WIDTH;
+					else total += s;
 				}
+				return total; 
 			}
 			Field(f) => {
 				var decl = env.lookupVar(f.ident.name.image, 0);
 				match (decl) {
 					x: PackingParamDecl => {
 						if (x.bitPos > 0) errAtToken(f.ident.name).PackingParamReused(f.ident.name.image);
 						else x.bitPos = x.bitWidth;
+						return x.bitWidth;
 					}
 					_ => errAtToken(f.ident.name).UnresolvedIdentifier(f.ident.name); // XXX: better error message
 				}
 			}
+			App(p, args) => { // TODO
+				var packingFn = packings[p.ident.name.image];
+				if (packingFn.pparams.length() != args.length()) {
+					errAtRange(args.range()).ArityMismatch(p.ident.name.image, packingFn.pparams.length(), args.length());
+				} else {
+					var pp = packingFn.pparams.list;
+					for (l = args.list; l != null; l = l.tail) {
+						var param = pp.head, paramSize = byte.!(tryUnboxPositiveInt(param.size, -1));
+						checkPackingExpr(env, l.head, paramSize); // TODO: proper size
+						pp = pp.tail;
+					}
+				}
+				return byte.!(tryUnboxPositiveInt(packingFn.size, -1));
```

Comment:
Similar here; we need to avoid casts so as not to crash the compiler for programs with type errors.